internal/state/conditions/conditions.go

```diff
@@ -315,3 +313,61 @@ func NewDefaultGatewayClassConditions() []Condition {
 		},
 	}
 }
+
+// NewDefaultGatewayCondition returns the default Condition that must be present in the status of a Gateway.
+func NewDefaultGatewayCondition() Condition {
```

Comment:
for other resource, we return []Condition in new Default function. Can we also do that for consistency?